src/api/db/migrate/20190520130009_create_commit_activities.rb

```diff
@@ -0,0 +1,16 @@
+class CreateCommitActivities < ActiveRecord::Migration[5.2]
+  # rubocop:disable Rails/CreateTableWithTimestamps
+  def change
+    create_table :commit_activities, options: 'CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci ROW_FORMAT=DYNAMIC', id: :integer do |t|
+      t.date :date, null: false
+      t.references :user, null: false, type: :integer
+      t.string :project, null: false, limit: 191
+      t.string :package, null: false, limit: 191
```

Comment:
It's `191` because the charset is `utf8mb4`, it still needs to fit in the [limit of 767 bytes set by the InnoDB engine](https://dev.mysql.com/doc/refman/8.0/en/charset-unicode-conversion.html).

Quote from the link
> InnoDB has a maximum index length of 767 bytes for tables that use COMPACT or REDUNDANT row format, so for utf8mb3 or utf8mb4 columns, you can index a maximum of 255 or 191 characters, respectively. If you currently have utf8mb3 columns with indexes longer than 191 characters, you must index a smaller number of characters. 

We do use `DYNAMIC` for the `ROW_FORMAT`, so we could use more, but I don't think it's needed.

Again the documentation:

> _Note_
> For InnoDB tables that use COMPRESSED or DYNAMIC row format, index key prefixes longer than 767 bytes (up to 3072 bytes) are permitted. Tables created with these row formats enable you to index a maximum of 1024 or 768 characters for utf8mb3 or utf8mb4 columns, respectively

Why do I know this? I learned the hard way in a previous job :smile:

Reply:
Ah, good to know:thumbsup:

Then this could cause conversion issues, or not? Project and package name are `utf8 COLLATE utf8_bin`

Reply:
actually I was confused myself cause it should have worked with the options given. We may have another look

Reply:
the problem is the dev container - it contains mariadb 10.0 (SLE12) from 2014 (first stable release), while in production we already use 10.2 (SLE15) from 2017. And you only get conversion problems between utf8mb4 and utf8 if you used emojis - I sure hope we have validations preventing that :)

Reply:
fixed